src/netconf_subscribed_notifications.c

```diff
@@ -2104,8 +2368,8 @@ np2srv_oper_sub_ntf_subscriptions_cb(sr_session_ctx_t *session, uint32_t UNUSED(
     char *name = NULL;
     uint32_t id;
 
-    if (strstr(request_xpath,"subscriptions/subscription[")
-		    && strstr(request_xpath,"/receivers/receiver[")) {
+    if (strstr(request_xpath, "subscriptions/subscription[") &&
+            strstr(request_xpath, "/receivers/receiver[")) {
```

Comment:
Not really part of this PR but this is wrong, the `request_path` can be, for example, `/*` and then you should provide all the data including the receivers. You should, instead, subscribe `np2srv_oper_sub_ntf_receivers_cb()` separately for its own nested subtree. Sysrepo will handle this correctly and call `np2srv_oper_sub_ntf_subscriptions_cb()` before `np2srv_oper_sub_ntf_receivers_cb()` as needed.

Reply:
I am not sure how to solve this one.
During the tests I see that np2srv_oper_sub_ntf_subscriptions_cb()
can give the operational data of everything in /subscriptions.
However when doing the reset action on a receiver, sysrepo calls the operational callback and  does not accept the complete operational data. But only the receiver operational data.
That's why I used this call to np2srv_oper_sub_ntf_receivers_cb() in this case.
What should np2srv_oper_sub_ntf_receivers_cb() do when it is called with request_xpath "/subscriptions" "or "/*" ?

Reply:
Okay, so I have finally looked more into the changes you made and actually understood at least the config/oper data. Frankly, the current relevant code is a mess and will need major improvements, so a long description of the changes is following.

The data tree `/ietf-subscribed-notifications:subscriptions/subscription` is supposed to hold information about both dynamic and configured subscriptions. Since there were only dynamic subscriptions before, I have implemented it only as operational data since they were based on RPC calls and not configuration data. However, that has changed with the configured subscriptions. Here is how I would put all this together:
- create a new config subscription (`sr_module_change_subscribe()`) for the whole subtree (you have implemented this in the callback `np2srv_config_subscriptions_cb()`)
- create a new config subscription for the receivers (your callback `np2srv_config_subscriptions_receivers_cb()`)
- create a new oper subscription to populate the state data in `/ietf-subscribed-notifications:subscriptions/subscription` for configured subscriptions
  - there are several ways of doing this, either subscribe for providing `/ietf-subscribed-notifications:subscriptions` (the callback will be called once) and create all the state data or subscribe to `/ietf-subscribed-notifications:subscriptions/subscription` (will be called for each configured subscription) and in the callback create data only for the specific subscription
  - use `SR_SUBSCR_OPER_MERGE` to only merge these data to the configuration data
- create a new config subscription for `/ietf-subscribed-notifications:subscriptions/ietf-subscribed-notif-receivers:receiver-instances` and store the data in internal structures (your callback `np2srv_config_receivers_cb()`)
- keep the oper subscription for `/ietf-subscribed-notifications:subscriptions` that creates the data for dynamic subscriptions exactly as it was but they should be merged (flag `SR_SUBSCR_OPER_MERGE`) to the data of the configured subscriptions
- regarding the internal structures holding all the information, it may make sense to completely separate dynamic and configured subscriptions because they are quite different, obviously

If you have any additional questions, we may schedule a call and talk about the design in detail.